bulldozer/signals.go

```diff
@@ -21,109 +21,389 @@ import (
 	"strings"
 
 	"github.com/palantir/bulldozer/pull"
+	"github.com/pkg/errors"
 	"github.com/rs/zerolog"
 )
 
+type Signal []string
+
 type Signals struct {
-	Labels            []string `yaml:"labels"`
-	CommentSubstrings []string `yaml:"comment_substrings"`
-	Comments          []string `yaml:"comments"`
-	PRBodySubstrings  []string `yaml:"pr_body_substrings"`
-	Branches          []string `yaml:"branches"`
-	BranchPatterns    []string `yaml:"branch_patterns"`
+	Labels            Signal `yaml:"labels"`
+	CommentSubstrings Signal `yaml:"comment_substrings"`
+	Comments          Signal `yaml:"comments"`
+	PRBodySubstrings  Signal `yaml:"pr_body_substrings"`
+	Branches          Signal `yaml:"branches"`
+	BranchPatterns    Signal `yaml:"branch_patterns"`
+	MaxCommits        int `yaml:"max_commits"`
 }
 
-func (s *Signals) Enabled() bool {
+type SignalsMatches struct {
+	Signals
+}
```

Comment:
I'm not sure I see the need for this type. It seems to have no extra methods and is not used as a parameter or return value, so the current use could probably use `Signals` directly (but as noted below, I'm not sure that's needed either.)

Reply:
Good point, removed